src/Module.ts

```diff
@@ -272,6 +272,13 @@ export default class Module {
 			props.frame = getCodeFrame(this.originalCode, location.line, location.column);
 		}
 
+		if (this.graph.phase < BuildPhase.GENERATE) {
+			const watchFiles = Object.keys(this.graph.watchFiles);
+			if (watchFiles.length > 0) {
+				props.watchFiles = watchFiles;
+			}
+		}
+
```

Comment:
I agree that adding the watchFiles to the error is probably the best solution, at least I could not think of a better approach. But instead of limiting it to the case where an explicit file position is provided, I changed the logic so that it triggers for every error in the build phase where there are added files, as in all those cases, the watcher will not receive any new watchFiles.

Reply:
Actually I changed it to "every error" as errors from the generate phase never seem to be triggered via this logic.